src/solvers/td3.ml

```diff
@@ -227,7 +230,9 @@ module WP =
         if tracing then trace "sol2" "init %a\n" S.Var.pretty_trace x;
         if not (HM.mem rho x) then (
           new_var_event x;
-          HM.replace rho x (S.Dom.bot ())
+          if init_bot_all || (init_bot_rhs && S.system x <> None) then
+            HM.replace rho x (S.Dom.bot ())
+          (* TODO: fix new_var_event repeatedly triggering when init-ing multiple times without value *)
```

Comment:
it looks like the solver variable counting happens very primitively by incrementing a counter:
https://github.com/goblint/analyzer/blob/5a434cdd4faab2d3a711d246d71f9b27f043f51d/src/solvers/generic.ml#L207-L209
Seems to me like this could just be `HM.length rho`, which would fix the new issue.